defold-spine/editor/src/spineext.clj

```diff
@@ -200,44 +200,52 @@
 ;;                                           (.transform transform p)
 ;;                                           [(.x p) (.y p) (.z p)])))))))
 
+
 (shader/defshader spine-id-vertex-shader
+  (uniform mat4 view_proj)
   (attribute vec4 position)
   (attribute vec2 texcoord0)
+  (attribute float page_index)
   (varying vec2 var_texcoord0)
+  (varying float var_page_index)
   (defn void main []
-    (setq gl_Position (* gl_ModelViewProjectionMatrix position))
-    (setq var_texcoord0 texcoord0)))
+    (setq gl_Position (* view_proj (vec4 position.xyz 1.0)))
+    (setq var_texcoord0 texcoord0)
+    (setq var_page_index page_index)))
 
 (shader/defshader spine-id-fragment-shader
   (varying vec2 var_texcoord0)
-  (uniform sampler2D texture_sampler)
+  (varying float var_page_index)
   (uniform vec4 id)
+  (uniform sampler2DArray texture_sampler)
   (defn void main []
-    (setq vec4 color (texture2D texture_sampler var_texcoord0.xy))
+    (setq vec4 color (texture2DArray texture_sampler (vec3 var_texcoord0 var_page_index)))
     (if (> color.a 0.05)
       (setq gl_FragColor id)
       (discard))))
 
 (def spine-id-shader (shader/make-shader ::id-shader spine-id-vertex-shader spine-id-fragment-shader {"id" :id}))
 
-(vtx/defvertex vtx-pos-tex-col
+; See SpineVertex
+(vtx/defvertex vtx-pos-tex-col-index
```

Comment:
Need to make sure that the sizes of each vertex are the same, otherwise the reading will be offset when rendering, and the mesh will not work properly.